src/clarity-angular/popover/popover.directive.ts

```diff
@@ -47,14 +55,23 @@ export class PopoverDirective {
         //inside of the popover. For Eg: Button Groups
         embeddedViewRef.detectChanges();
 
-        this._popoverInstance = new Popover(embeddedViewRef.rootNodes[0]);
-        this._popoverInstance.anchor(
-            this.anchorElem, this.anchorPoint, this.popoverPoint, this.popoverOptions);
+        // filter out other nodes in the view ref so we are only left with element nodes
+        let elementNodes: HTMLElement[] = embeddedViewRef.rootNodes.filter((node: any) => {
+           return node.nodeType === 1;
+        });
```

Comment:
Yes.. depending on the syntax we use, the rootNodes could contain just the element we want to use, or contain textnodes before and after.. so it became unreliable to try to pass in a specific node with something like `rootNodes[i]` (where `i` would be either 0 or 1 in the cases that I've seen so far).